packages/jsii-dotnet-runtime/src/Amazon.JSII.Runtime/ReflectionUtils.cs

```diff
@@ -75,5 +76,71 @@ public static JsiiClassAttribute GetClassAttribute(Type type)
             return type.GetCustomAttribute<JsiiClassAttribute>()
                    ?? GetClassAttribute(type.BaseType);
         }
+        
+        /// <summary>
+        /// Looks for a Method attribute, starting from the current class hierarchy, and searching on implemented
+        /// interfaces if none was found.
+        /// </summary>
+        /// <param name="method">The method for which an attribute is needed</param>
+        /// <param name="inherit">Whether inherited declarations should be searched for (if false, interfaces will NOT be searched)</param>
+        /// <typeparam name="T">The attribute type that is desired</typeparam>
+        /// <returns>The attribute if one was found, or null</returns>
+        [return: MaybeNull]
+        internal static T GetAttribute<T>(this MethodInfo method, bool inherit = true) where T : Attribute
+        {
+            var directAttribute = method.GetCustomAttribute<T>(inherit);
+            if (directAttribute != null || !inherit) return directAttribute;
+            
+            var type = method.ReflectedType;
+            if (type == null)
+            {
+                return null;
+            }
+
+            var interfaceMethods = type.GetTypeInfo().ImplementedInterfaces
+                .Select(ii => type.GetInterfaceMap(ii))
+                .SelectMany(map =>
+                    Enumerable.Range(0, map.TargetMethods.Length)
+                        .Where(n => map.TargetMethods[n] == method)
+                        .Select(n => map.InterfaceMethods[n]));
+            
+            return interfaceMethods
+                .Select(im => im.GetCustomAttribute<T>(true))
+                .FirstOrDefault(attribute => attribute != null);
+        }
+        
+        /// <summary>
+        /// Looks for a Property attribute, starting from the current class hierarchy, and searching on implemented
+        /// interfaces if none was found.
+        /// </summary>
+        /// <param name="property">The property for which an attribute is needed</param>
+        /// <param name="inherit">Whether inherited declarations should be searched for (if false, interfaces will NOT be searched)</param>
+        /// <typeparam name="T">The attribute type that is desired</typeparam>
+        /// <returns>The attribute if one was found, or null</returns>
+        [return: MaybeNull]
+        internal static T GetAttribute<T>(this PropertyInfo property, bool inherit = true) where T : Attribute
```

Comment:
These 2 methods look way too similar to me. Can this not be factored out with some generics?